ballerina/http_log_manager.bal

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+// TODO: Make the fields optional once configurable records support union members
+# Represents HTTP trace log configuration.
+#
+# + console - Boolean value to enable or disable console trace logs
+# + path - File path to store trace logs
+# + host - Socket hostname to publish the trace logs
+# + port - Socket port to publish the trace logs
+public type TraceLogConfiguration record {|
+    boolean console = false;
+    string path = "";
+    string host = "";
+    int port = 0;
```

Comment:
Can't we make these optional fields ?

Reply:
Currently union member type **()**  is not supported in **configurable** records. Once the support is given we can change these fields to optional.
```ballerina
public type TraceLogConfiguration record {|
    boolean consoleEnabled = false;
    string? path = ();
    string? host = ();
    int? port = ();
|};

configurable TraceLogConfiguration & readonly traceLogConfig = {};
```
The above code returns a compile time error : 
```
configurable variable currently not supported for '(TraceLogConfiguration & readonly)' 
union member type '()' is not supported
```

Reply:
I think the limitation is there for `nillable` fields. But you could use optional fields.
```ballerina
public type TraceLogConfiguration record {|
    boolean consoleEnabled = false;
    string path?;
    string host?;
    int port?;
|};

configurable TraceLogConfiguration & readonly traceLogConfig = {};
```

Reply:
+1 Shall we do this change and see? If so we can get rid of default values